### ISSUE TITLE
add level of detail mode to octree

### ISSUE THREAD

**Issue by agurvich** [ageller/Firefly#117]
an extension to #116, what if instead of having a single `.ffly` file associated with each node there were multiple levels of detail `.ffly` files with smaller and smaller decimations. rather than opening/closing nodes we would move them up/down in LoD. would need to keep track of each node's LoD (kind of easy) and then determine when to load additional files for each node (very hard). 

would need:
* associating node particles with > 1 `.ffly` file (currently assumes all node buffer particles live in a single `.ffly` file
* associating multiple particle meshes with a single node
* new "opening/closing" criteria
* rewrite octree walk to walk until threshold LoD rather than until it hits a closed node

so in short, a lot of work. could be a good ~week or two project but should not hold up submission.
